### PR TITLE
Feature/repeat order

### PR DESCRIPTION
**Pull request by AnirupPat**
This PR contains the following feature :
- Repeat order functionality

### CODE REVIEW

File: src/app/components/home/home.component.ts [hunk 1]
```diff
@@ -36,6 +36,8 @@ export class HomeComponent implements OnInit {
   }
 
    ngOnInit() {
+    console.log(Common.getIdBasedQueryString(["1","3","5"]))
```
Comment (AnirupPat) [inline src/app/components/home/home.component.ts]: @reddy-s  let this be there...i will remove it when i get rid of it

File: src/app/components/orders/orders.component.ts [hunk 2]
```diff
@@ -161,4 +172,51 @@ export class OrdersComponent implements OnInit {
       this.ngZone.run(() =>this.router.navigate(['review/create/product', productId] )).then()
     }
   }
-}
+
+  repeatOrder(selectedOrder) {
+    var cart = [];
+    console.log(selectedOrder)
```
Comment (reddy-s) [inline src/app/components/orders/orders.component.ts]: This is think can go away

File: src/app/components/orders/orders.component.ts [hunk 3]
```diff
@@ -161,4 +172,51 @@ export class OrdersComponent implements OnInit {
...
+  }
+
+  // method to post items to the cart..
+   postCartItems() {
+    return new Promise( (resolve, reject) => {
+      this._loginStateService.loaderEnable()
+    let items = [];
+      for(var i = 0; i < this.newItemsToCart.length; i++) {
+        items.push(this.updateCartItem(this.newItemsToCart[i].id, this.newItemsToCart[i].quantity, true))
+      }
```
Comment (reddy-s) [inline src/app/components/orders/orders.component.ts]: can you not use 
```suggestion
      this.newItemsToCart.foreach((item) => { 
          items.push(this.updateCartItem(item.id, item.quantity.quantity, true))
      })
```

File: src/app/components/orders/orders.component.ts [hunk 4]
```diff
@@ -161,4 +172,51 @@ export class OrdersComponent implements OnInit {
       this.ngZone.run(() =>this.router.navigate(['review/create/product', productId] )).then()
     }
   }
-}
+
+  repeatOrder(selectedOrder) {
+    var cart = [];
+    console.log(selectedOrder)
+    for(let order in this.orders[selectedOrder]['items']) {
```
Comment (reddy-s) [inline src/app/components/orders/orders.component.ts]: You can use foreach even here

File: src/app/utils/common.ts [hunk 5]
```diff
@@ -91,5 +91,14 @@ export class Common {
       // updatedQuery.q = updatedQuery.searchedText || updatedQuery.q
     return updatedQuery
   }
-  
+
+  public static getIdBasedQueryString = (_productIds: string[]) => {
+    let _queryItem = "(term+field=_id+{p})"
+    let _queryString: string = "(or+";
+    _productIds.forEach((productId) => {
+    _queryString = _queryString + _queryItem.replace("{p}", productId)
```
Comment (reddy-s) [inline src/app/utils/common.ts]: indentation is not right here

File: src/app/components/orders/orders.component.ts (hunk 2 above)
Comment (AnirupPat) [reply src/app/components/orders/orders.component.ts]: yes..

File: src/app/components/orders/orders.component.ts (hunk 3 above)
Comment (AnirupPat) [reply src/app/components/orders/orders.component.ts]: done !

File: src/app/utils/common.ts (hunk 5 above)
Comment (AnirupPat) [reply src/app/utils/common.ts]: done !

File: src/app/components/orders/orders.component.ts (hunk 4 above)
Comment (AnirupPat) [reply src/app/components/orders/orders.component.ts]: done !

**Pull request merged**